packages/models/src/store/json-storage.ts

```diff
@@ -0,0 +1,87 @@
+import { ChainStorage, StorageOnChain, StorageOffChain } from './chain-storage'
+
+type JSONStorageType = string | number | boolean | bigint
+export type JSONStorageOffChain =
+  | JSONStorageType
+  | {
+      [x: string]: JSONStorageType | JSONStorageType[] | JSONStorageOffChain
```

Comment:
Using `Record` makes it concise

Reply:
`Record` will throw an error named circular reference.
<img width="635" alt="image" src="https://user-images.githubusercontent.com/12881040/206210926-61d8fc33-2844-4bf3-8199-81f75588074a.png">

Reply:
Got it